runtime/lua/vim/_editor.lua

```diff
@@ -1042,50 +1042,75 @@ end
 ---
 ---@return string|nil # Deprecated message, or nil if no message was shown.
 function vim.deprecate(name, alternative, version, plugin, backtrace)
```

Comment:
Although it's not very important, the behavior of `vim._deprecate` (plugin = 'Nvim') and `vim.deprecate` (for plugins?) are different. I wonder if we can make it more consistent while making deprecation warnings less annoying as intended.

Reply:
We could change vim.deprecate so plugins use the new deprecation functionality as well.. Do you think that's the way to go?

Reply:
If we go with this PR, then plugins should use the same way of showing deprecation messages as Nvim. Though `vim.deprecate` no longer benefits from `vim.notify` and `vim.notify_once`.

Reply:
The `defaults to "Nvim"` in the comment above is no longer true.

Reply:
> If we go with this PR, then plugins should use the same way of showing deprecation messages as Nvim. Though `vim.deprecate` no longer benefits from `vim.notify` and `vim.notify_once`.

I'm not sure they should. I think a problem with vim.deprecate is that we made it public too soon and it just grew without much consideration. I think giving the internal deprecation mechanism to mature a bit would help us more in the long run.